kernel/src/misc/wm/wm.c

```diff
@@ -435,14 +435,25 @@ wm_window_t* wm_window_at(int32_t x, int32_t y) {
 }
 
 void wm_draw_mouse(rect_t old, rect_t new) {
-    wm_refresh_partial(old);
+
+
+    if((old.top != new.top) || (old.left != new.left))
+    	wm_refresh_partial(old);
 
     uintptr_t addr = fb.address + new.top*fb.pitch + new.left*fb.bpp/8;
 
-    for (int32_t y = 0; y < new.bottom - new.top; y++) {
-        memset((void*) addr, 127, (new.right - new.left)*fb.bpp/8);
+    for (int32_t y = 0; y < new.bottom - new.top - 6; y++) {
+        memset((void*) addr, 127, (y+1)*fb.bpp>>3);
         addr += fb.pitch;
     }
+
+    addr += 4*fb.bpp>>3;
+
+    for (int32_t y = 0; y < 6; y++) {
+        memset((void*) addr, 127, 3*fb.bpp>>3);
+	addr += fb.bpp>>3;
+	addr += fb.pitch;
```

Comment:
About using a shift, I checked and on -O3 optimisation level, gcc produces the same `shr`-based code as with a division by 8. So all things being equal, I'd rather keep using `fb.bpp/8` for clarity.

Also there's a space/tabs thing going on here, could you make sure indentation is four spaces?

Reply:
You know, I thought about that.  I reverted the shifts as requested and adjusted tabs back to spaces.